README.md

```diff
@@ -1,102 +1,269 @@
+has multiple backends. Currently: "piet-direct2d", "piet-cairo", and "piet-web".
+Currently the macOS and Linux versions of Druid use the piet-cairo backend,
+while Windows uses piet-direct2d. A GPU backend is planned. Here's a simple
+drawing example from the [custom_widget] example.
 
- ```shell
-$> PKG_CONFIG_PATH="/usr/local/opt/libffi/lib/pkgconfig" cargo build
- ```
-#### Linux
+```rust
+// Create an arbitrary bezier path
+// (base_state.size() returns the size of the layout rect we're painting in)
+let mut path = BezPath::new();
+path.move_to(Point::ORIGIN);
+path.quad_to(
+    (80.0, 90.0),
+    (base_state.size().width, base_state.size().height),
+);
+// Create a color
+let stroke_color = Color::rgb8(0x00, 0x80, 0x00);
+// Stroke the path with thickness 1.0
+paint_ctx.stroke(path, &stroke_color, 1.0);
+
+// Rectangles: the path for practical people
+let rect = Rect::from_origin_size((10., 10.), (100., 100.));
+// Note the Color:rgba8 which includes an alpha channel (7F in this case)
+let fill_color = Color::rgba8(0x00, 0x00, 0x00, 0x7F);
+paint_ctx.fill(rect, &fill_color);
+```
+
+### widgets
+
+Widgets in druid are objects which implement the `Widget` trait. The trait is
+parametrized by a type (`T`) for associated data. All trait methods are provided
+with access to this data, and in the case of `event` the reference is mutable,
+so that events can directly update the data.
+
+Whenever the application data changes, the framework traverses the widget
+hierarchy with an `update` method. 
+
+All the widget trait methods are provided with a corresponding context
+([EventCtx], [LayoutCtx], [PaintCtx], [UpdateCtx]). The widget can request
+things and cause actions by calling methods on that context.
+
+In addition, all trait methods are provided with an environment `Env`, which
+includes the current theme parameters (colors, dimensions, etc.).
+
+```rust
+impl<T: Data> Widget<T> for Button<T> {
+    fn paint(&mut self, paint_ctx: &mut PaintCtx, base_state: &BaseState, data: &T, env: &Env) {
+      ...
+    }
+    fn layout(
+        &mut self,
+        layout_ctx: &mut LayoutCtx,
+        bc: &BoxConstraints,
+        data: &T,
+        env: &Env,
+    ) -> Size {
+      ...
+    }
+    fn event(&mut self, event: &Event, ctx: &mut EventCtx, data: &mut T, env: &Env) {
+      ...
+    }
+    fn update(&mut self, ctx: &mut UpdateCtx, old_data: Option<&T>, data: &T, env: &Env) {
+      ...
+    }
+}
+```
+
+Druid provides a number of [basic utility and layout widgets] and it's easy to
+implement your own. You can also compose widgets into new widgets:
+
+```rust
+fn build_widget() -> impl Widget<u32> {
+    let mut col = Column::new();
+    for i in 0..30 {
+        let button = Button::new(format!("Button {}", i), Button::noop);
+        col.add_child(Padding::new(3.0, button), 0.0);
+    }
+    Scroll::new(col)
+}
+```
+
+### layout
+
+Druid's layout protocol is [strongly inspired by Flutter]. In their `layout`
```

Comment:
I initially thought that `their` was referring to Flutter. But then I thought maybe it refers to druid widgets. Since there's some explanation here, I would just be surprised if the `layout` section talked more about Flutter than druid.

Reply:
Druid's layout protocol is strongly inspired by [Flutter's box layout model]. In druid, widgets are passed a `BoxConstraint` that provides them a minimum and
maximum size for layout.

Reply:
Sounds good.